### PR TITLE
Feature/get events from categories

### LINKED ISSUE #352: Add helper to select events from one category
It would be nice to have some helper function that automatically selects events given a config category based on an existing `category_ids`column.

```
masked_events = columnflow.columnar_util.select_category(events, config.get_category("1e"))
```

This should automatically select all events that contain a category id of one of the "1e" leaf categories.

### PR DESCRIPTION
**Pull request by mafrahm**
Closes #352

### CODE REVIEW

File: columnflow/config_util.py [hunk 1]
```diff
@@ -25,8 +25,50 @@
...
+
+
+def get_events_from_categories(
+    events: ak.Array,
+    categories: Sequence[str | od.Category],
+    config_inst: od.Config | None = None,
+) -> ak.Array:
+    """
+    Helper function that returns all events from an awkward array *events* that are categorized
+    into one the leafs of one of the *categories*.
```
Comment (pkausw) [inline columnflow/config_util.py:42]:
```suggestion
    into one of the leafs of one of the *categories*.
```

File: tests/test_config_util.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        leaf_22 = main_2.add_category("leaf_22", id=22)
+        leaf_22.add_category("leaf_221", id=221)
+
+        # awkward array with category_ids (only leaf ids)
+        self.events = ak.Array({
+            "category_ids": [[1], [21, 221], [21], [221]],
+            "dummy_field": [1, 2, 3, 4],
+        })
+
+    def test_get_events_from_categories(self):
```
Comment (pkausw) [inline tests/test_config_util.py:40]: You could probably split this test into each of the different blocks (i.e. smaller unit tests), but this is optional imho

File: tests/test_config_util.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        # check that leaf category is working
+        events = get_events_from_categories(self.events, ["main_1", "main_2"], self.config_inst)
+        self.assertTrue(ak.all(events.dummy_field == ak.Array([1, 2, 3, 4])))
+
+        # check that directly passing category inst is working
+        events = get_events_from_categories(self.events, self.config_inst.get_category("main_1"))
+        self.assertTrue(ak.all(ak.any(events.category_ids == 1, axis=1)))
+        self.assertTrue(ak.all(events.dummy_field == ak.Array([1])))
+
```
Comment (pkausw) [inline tests/test_config_util.py:63]: The new helper also `raises` an Exception at some point, would also be good to test this explicitely

**Review by pkausw — CHANGES_REQUESTED** [review]
Nice PR @mafrahm ! I think it would be nice to further test things that are not supposed to work, both to make sure that unexpected behaviors are less probable and also for more clarity of how this module is supposed to work

**Review by riga — APPROVED** [review]
LGTM!

**Pull request merged**